spec/index.bs

```diff
@@ -2179,12 +2211,83 @@ interface GPUTextureUsage {
+                                </dl>
+                            - If |descriptor|.{{GPUTextureDescriptor/sampleCount}} > 1 (|descriptor|.{{GPUTextureDescriptor/sampleCount}} is 4):
+                                - |descriptor|.{{GPUTextureDescriptor/mipLevelCount}} must be 1.
+                                - |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/depthOrArrayLayers=] must be 1.
+                                - |descriptor|.{{GPUTextureDescriptor/usage}} must not include the {{GPUTextureUsage/STORAGE}} bit.
+                                - |descriptor|.{{GPUTextureDescriptor/format}} must be a [=renderable format=].
+                            - |descriptor|.{{GPUTextureDescriptor/mipLevelCount}} must be less than or equal to
+                                [$maximum mipLevel count$](|descriptor|.{{GPUTextureDescriptor/dimension}}, |descriptor|.{{GPUTextureDescriptor/size}}).
+                            - |descriptor|.{{GPUTextureDescriptor/usage}} must be a combination of {{GPUTextureUsage}} values.
+                            - If |descriptor|.{{GPUTextureDescriptor/usage}} includes the {{GPUTextureUsage/RENDER_ATTACHMENT}} bit,
+                                |descriptor|.{{GPUTextureDescriptor/format}} must be a [=renderable format=].
+                            - If |descriptor|.{{GPUTextureDescriptor/usage}} includes the {{GPUTextureUsage/STORAGE}} bit,
+                                |descriptor|.{{GPUTextureDescriptor/format}} must be listed in [[#plain-color-formats]] table
+                                with {{GPUTextureUsage/STORAGE}} capability.
+                            - Some |descriptor|.{{GPUTextureDescriptor/format}}s require [=feature=]s when creating a {{GPUDevice}}:
+                                - {{GPUTextureFormat/"depth24unorm-stencil8"}} requires [[#depth24unorm-stencil8]].
+                                - {{GPUTextureFormat/"depth32float-stencil8"}} requires [[#depth32float-stencil8]].
+                                - BC compressed format requires [[#texture-compression-bc]].
+                            - If |descriptor|.{{GPUTextureDescriptor/format}} is a compressed format:
+                                - |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/width=] must be multiple of [=texel block width=].
+                                - |descriptor|.{{GPUTextureDescriptor/size}}.[=Extent3D/height=] must be multiple of [=texel block height=].
+                        </div>
+
+                        Then:
+                            1. Generate a {{GPUValidationError}} in the current scope with appropriate error message.
+                            1. Return a new [=invalid=] {{GPUTexture}}.
+
+                    1. Let |t| be a new {{GPUTexture}} object.
+                    1. Set |t|.{{GPUTexture/[[textureSize]]}} to |descriptor|.{{GPUTextureDescriptor/size}}.
```

Comment:
we should just grab the whole descriptor here instead of copying all the fields

Reply:
They are the internal slots of a texture. And there is no something like `TextureDescriptor` as a whole for all these internal slots. So I guess we should assign these internal slots one by one? I noticed that other GPUObjects assign internal slots in the same way: assign internal slots one by one. So it should be OK?

Reply:
we can have an internal slot of type `GPUTextureDescriptor`

Reply:
Good idea. We can have a `GPUTextureDescriptor` because the current internal slots are exactly the members of a `GPUTextureDescriptor` object. However, a few other places (Texture view validation, texture copy validation, etc.) refer to the current internal slots directly. So, can we leave that change to a separate PR?

Reply:
Yes, I don't want to have to update all the refs in this PR, better to have it separate.

Reply:
OK. I will submit a separate PR right after this one is merge.

Reply:
thank you!

Reply:
Any other concerns, @kvark (and @kainino0x )? I think this PR is ready to go. Please help to merge if it is OK.